lib/elixir/test/elixir/kernel/errors_test.exs

```diff
@@ -648,6 +650,17 @@ defmodule Kernel.ErrorsTest do
     assert_compile_fail CompileError,
       "nofile:1: cannot use ^x outside of match clauses",
       'x = 8; <<a, b::size(^x)>> = <<?a, ?b>>'
+
+    warning = capture_io(:stderr, fn ->
```

Comment:
Looks like this test belongs to `Kernel.WarningTest`, otherwise we lose `async: true` property.